src/modules/px4iofirmware/safety_button.cpp

```diff
@@ -122,7 +112,7 @@ safety_check_button(void *arg)
 	/* Select the appropriate LED flash pattern depending on the current IO/FMU arm state */
 	uint16_t pattern = LED_PATTERN_FMU_REFUSE_TO_ARM;
 
-	if (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_OFF) {
+	if (r_status_flags & PX4IO_P_STATUS_FLAGS_SAFETY_BUTTON_PRESSED) {
```

Comment:
This should represent the 'safety off'  state, not when the button is pressed

Reply:
`safety_off` is maybe a bit confusing to keep in px4io as you can see comment https://github.com/PX4/PX4-Autopilot/pull/19558#discussion_r864168326
also now I understand that `PRESSED` is also confusing could be any moment between 0 and 1 second while the button is pressed. What do you think about replacing it with this:
`#define PX4IO_P_STATUS_FLAGS_SAFETY_BUTTON_EVENT	(1 << 10) /* state indicating event when the safety button is pressed for 1 second*/`

If you still think` safety off` is better I can revert it.

Reply:
I wasn't refering to the naming, but to the functional change: before it meant 'safety button was pressed, and IO is now ready to arm', now it means 'safety button is currently being pressed' (condition is cleared after the ACK).
Not sure how important it is to keep it, but I think it's worth keeping as to give clear user indication. For that you need to pass the `safety_off` state to the IO.

Reply:
Thanks for the additional clarification! I am passing `safety_off` to PX4IO now for this purpose.